Convert draw rectangle gallery example into a how-to guide (#7435)
sunpy/sunpy: docs/how_to/create_rectangle_on_map.rst

.. _sunpy-how-to-create-rectangle-on-map:

************************************
How to Draw a Rectangle on sunpy map
************************************

> SunPy provides a convenient method called :meth:`~sunpy.map.GenericMap.draw_quadrangle` to draw rectangles on maps.

Reviewer: ```suggestion
``sunpy`` provides a convenient method called :meth:`~sunpy.map.GenericMap.draw_quadrangle` to draw rectangles on maps.
```

> In this guide, we'll demonstrate four different ways to draw a rectangle on a map using SunPy.

Reviewer: ```suggestion
In this guide, we will demonstrate four different methods to draw a rectangle on a `sunpy.map.Map`.
```

Specify Corners as a Single `~astropy.coordinates.SkyCoord` Object
==================================================================

Taking two opposite corners as a single SkyCoord object

.. code-block:: python

    >>> import matplotlib.pyplot as plt

    >>> import astropy.units as u
    >>> from astropy.coordinates import SkyCoord

    >>> import sunpy.data.sample
    >>> import sunpy.map

>     >>> aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)

Reviewer: This needs to be marked remote-data

Author: done

    >>> fig = plt.figure(figsize=(5, 5))
    >>> ax = fig.add_subplot(projection=aia_map)
    >>> aia_map.plot(axes=ax, clip_interval=(1, 99.99)*u.percent) # doctest: +SKIP
    >>> coords = SkyCoord(Tx=(100, 500) * u.arcsec, Ty=(200, 500) * u.arcsec,frame=aia_map.coordinate_frame)
    >>> aia_map.draw_quadrangle(coords, axes=ax, edgecolor="blue") # doctest: +SKIP
    >>> plt.show() # doctest: +SKIP

> Specify Corners as Separate `~astropy.coordinates.SkyCoord` Objects
> ===================================================================

Reviewer: ```suggestion
Specify corners with separate `~astropy.coordinates.SkyCoord`
=============================================================
```

> Taking two opposite corners as separate SkyCoord Objects

Reviewer: ```suggestion
We will use two `~astropy.coordinates.SkyCoord` to represent the two opposite corners.
```

.. code-block:: python

>     >>> aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
>     >>> fig = plt.figure(figsize=(5, 5))
>     >>> ax = fig.add_subplot(projection=aia_map)
>     >>> aia_map.plot(axes=ax, clip_interval=(1, 99.99)*u.percent) # doctest: +SKIP
>     >>> bottom_left = SkyCoord(-500 * u.arcsec, 200 * u.arcsec, frame=aia_map.coordinate_frame)
>     >>> top_right = SkyCoord(-100 * u.arcsec, 500 * u.arcsec, frame=aia_map.coordinate_frame)
>     >>> aia_map.draw_quadrangle(bottom_left, axes=ax, top_right=top_right, edgecolor="green") # doctest: +SKIP
>     >>> plt.show() # doctest: +SKIP

Reviewer: ```suggestion
    >>> aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)

    >>> fig = plt.figure(figsize=(5, 5))
    >>> ax = fig.add_subplot(projection=aia_map)
    >>> aia_map.plot(axes=ax, clip_interval=(1, 99.99)*u.percent)  # doctest: +SKIP

    >>> bottom_left = SkyCoord(-500 * u.arcsec, 200 * u.arcsec, frame=aia_map.coordinate_frame)
    >>> top_right = SkyCoord(-100 * u.arcsec, 500 * u.arcsec, frame=aia_map.coordinate_frame)
    >>> aia_map.draw_quadrangle(bottom_left, axes=ax, top_right=top_right, edgecolor="green")  # doctest: +SKIP

    >>> plt.show()  # doctest: +SKIP
```

> Specify One Corner, Width, and Height
> =====================================

Reviewer: ```suggestion
Specify one corner with a width and height
==========================================
```

.. code-block:: python

>     >>> aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
>     >>> fig = plt.figure(figsize=(5, 5))
>     >>> ax = fig.add_subplot(projection=aia_map)
>     >>> aia_map.plot(axes=ax, clip_interval=(1, 99.99)*u.percent) # doctest: +SKIP
>     >>> bottom_left = SkyCoord(-500 * u.arcsec, -500 * u.arcsec, frame=aia_map.coordinate_frame)
>     >>> width = 400 * u.arcsec
>     >>> height = 300 * u.arcsec
>     >>> aia_map.draw_quadrangle(bottom_left, axes=ax, width=width, height=height, edgecolor="yellow") # doctest: +SKIP
>     >>> plt.show() # doctest: +SKIP

Reviewer: ```suggestion
    >>> aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
    
    >>> fig = plt.figure(figsize=(5, 5))
    >>> ax = fig.add_subplot(projection=aia_map)
    >>> aia_map.plot(axes=ax, clip_interval=(1, 99.99)*u.percent)  # doctest: +SKIP
    
    >>> bottom_left = SkyCoord(-500 * u.arcsec, -500 * u.arcsec, frame=aia_map.coordinate_frame)
    >>> width = 400 * u.arcsec
    >>> height = 300 * u.arcsec
    >>> aia_map.draw_quadrangle(bottom_left, axes=ax, width=width, height=height, edgecolor="yellow")  # doctest: +SKIP
    
    >>> plt.show()  # doctest: +SKIP
```

> Draw a Rectangle in Pixel Coordinates
> =====================================

Reviewer: ```suggestion
Using pixel coordinates
-----------------------

We will use a `~astropy.coordinates.SkyCoord` to work out the pixel coordinates instead of using coordinates as we do above.
```

.. _sunpy-how-to-create-rectangle-on-map:

************************************
How to Draw a Rectangle on sunpy map
************************************

``sunpy`` provides a convenient method called :meth:`~sunpy.map.GenericMap.draw_quadrangle` to draw rectangles on maps.
In this guide, we will demonstrate four different methods to draw a rectangle on a `sunpy.map.Map`.

Specify corners with a single `~astropy.coordinates.SkyCoord`
=============================================================

We will use one `~astropy.coordinates.SkyCoord` to represent the two opposite corners.

> .. code-block:: python

Reviewer: So these have not rendered at all in the documentation: 

https://sunpy--7435.org.readthedocs.build/en/7435/how_to/create_rectangle_on_map.html

We will need to fix that.

Author: I am sorry but what do you exactly mean by not rendered in docs?

Reviewer: There is are figures

Author: Oh we need figures? Alrighty will do!
And do we need  4 figures or one containing all like we had in the example, with the legend?

Reviewer: Let's start with 4 separate ones and we can slim down after if we need to
